File: algorithms/Occurrences After Bigram/Occurrences After Bigram.java
class Solution {
    public String[] findOcurrences(String text, String first, String second) {
        List<String> list = new ArrayList<>();
        
        String[] arr = text.split(" ");
        
        for(int i = 0; i < arr.length - 2; i++) {
            if(arr[i].equals(first) && arr[i + 1].equals(second)) {
                list.add(arr[i + 2]);
            }
        }
        
        String[] result = list.toArray(new String[0]);
        
        return result;
    }
}

File: algorithms/O/Occurrences After Bigram/Occurrences After Bigram.cpp
class Solution {
public:
	vector<string> findOcurrences(string text, string first, string second) {
		vector<string>ans;
		int i=0;
		while(i<text.length())
		{
			string word = "";
			string secondWord="";
			while(i<text.length() && text[i]!=' ')
			{
				word = word+text[i];
				i++;
			}
			if(word == first)
			{
				int k = i+1;
				while(k<text.length() && text[k]!=' ')
				{
					secondWord = secondWord+text[k];
					k++;
				}  
				k++;
				if(k<text.length() && secondWord==second)
				{
					string tmp="";
					int j=k;
					while(j<text.length() && text[j]!=' ')
					{
						tmp = tmp+text[j];
						j++;
					} 

					ans.push_back(tmp);
				}
			}
		   i++;
		}
		return ans;
	}
};